### PR TITLE
python3Packages.sphinx-intl init at 2.1.0 and transifex-cli: init at 1.6.7

### PR DESCRIPTION
**Pull request by thornycrackers**
###### Description of changes

This PR packages up `sphinx-intl`. To get the tests passing I also had to package up `transifex-cli`. This is for packaging request #223863 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/transifex-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "sha256-5166P44HSRKQ0pCh1BCPd1ZUryh/IBDumcnLYA+CSBY=";
+  };
+
+  vendorSha256 = "sha256-rcimaHr3fFeHSjZXw1w23cKISCT+9t8SgtPnY/uYGAU=";
+
+  # https://github.com/transifex/cli/issues/170
+  ldflags = [
+    "-X 'github.com/transifex/cli/internal/txlib.Version=${version}'"
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/transifex-cli/default.nix:21]:
```suggestion
    "-s" "-w" "-X 'github.com/transifex/cli/internal/txlib.Version=${version}'"
```

File: pkgs/applications/misc/transifex-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "transifex";
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "sha256-5166P44HSRKQ0pCh1BCPd1ZUryh/IBDumcnLYA+CSBY=";
+  };
+
+  vendorSha256 = "sha256-rcimaHr3fFeHSjZXw1w23cKISCT+9t8SgtPnY/uYGAU=";
+
+  # https://github.com/transifex/cli/issues/170
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/transifex-cli/default.nix:19]:
```suggestion
```

File: pkgs/development/python-modules/sphinx-intl/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  propagatedBuildInputs = [
+    babel
+    click
+    setuptools
+    sphinx
+  ];
+
+  checkInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinx-intl/default.nix:32]:
```suggestion
  nativeCheckInputs = [
```

File: pkgs/development/python-modules/sphinx-intl/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    pytestCheckHook
+    mock
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinx-intl/default.nix:34]:
```suggestion
    mock
    transifex-cli
```

File: pkgs/development/python-modules/sphinx-intl/default.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+
+  pythonImportsCheck = [ "sphinx_intl" ];
+
+  preCheck = ''
+    export PATH=${lib.getBin transifex-cli}/bin:$PATH
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinx-intl/default.nix:41]:
```suggestion
```

**Comment by thornycrackers** [comment]
Thank you @SuperSandro2000 for the quick reply. I applied all the suggestions, cleaned up the commits, and tested that the package still builds correctly locally.

**Pull request merged**